libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
So drop the whole block then.  If that's getting triggered after `$this->connect()` then we have a logic flaw in the `connect()` method.